app/src/androidTest/java/org/kiwix/kiwixmobile/main/KiwixMainRobot.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.main
+
+import applyWithViewHierarchyPrinting
+import com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertDisplayed
```

Comment:
I'd like to remove our dependency on Barista as much as possible, if you can rewrite this to not use it then that would be great, if it is too difficult I can compromise easily

Reply:
I eliminated it entirely, but why don't you like it?

Reply:
a) not kotlin first
b) wraps espresso and I personally hate espresso
c) has very surprising rules that caused me no end of trouble when I started on this project
d) it really isn't that hard to write your own wrapper that suits the individual needs of your own project

Reply:
I have query only about the _a)_ part, if a library is written in Java then how does it fall back in comparison of its Kotlin equivalent? As ultimately we are going to use the functions/features provided by it(library), that will always remain same irrespective of the language used for building that lib.

Reply:
If a Library is written in kotlin and specifically targets kotlin then the kotlin features can shine, you can have default arguments, infix functions, extension functions, accept higher order functions, specify your api as top level functions, compiler supported null safety, create DSLs and so much more. All of this code would be callable from java but it would look very clumsy and bloated and be inherently more error prone. Barista is rewritten in kotlin to a large degree but its legacy as a java project is showing

Reply:
Fine, gotcha!